setup.py

```diff
@@ -40,9 +40,8 @@ def read(fname):
 gcs_deps = ['google-cloud-storage>=1.31.0']
 azure_deps = ['azure-storage-blob', 'azure-common', 'azure-core']
 http_deps = ['requests']
-daos_deps = ['pydaos']
 
-all_deps = aws_deps + gcs_deps + azure_deps + http_deps + daos_deps
+all_deps = aws_deps + gcs_deps + azure_deps + http_deps
```

Comment:
Why did you revert this? The previous commit makes sense to me.

Reply:
pydaos as a standalone library is not available. This dependency ends up with errors trying to auto install pydaos. The last set of automated tests failed. pydaos (dependency) gets installed with the DAOS package(rpm).

Reply:
Ah, I just saw your previous comment. Thank you for explaining this to me (again!).

If pydaos gets installed via an rpm (or otherwise package), then where do the files from the package actually go? I imagine it would be the system's Python, as opposed to the virtual environment. So, that means smart_open would have to be installed into the system's Python as well, right?

Reply:
Yes, that's right. By default, pydaos gets installed into system's Python. smart_open installed in the same system's python works fine. 
Example: 
/usr/lib64/python3.6/site-packages/pydaos
/usr/local/lib/python3.6/site-packages/smart_open

If there are different virtual environments created, then one has to link it up appropriately. 
The earlier comment on sys.path.append had the custom path for the pydaos library. pydaos documentation has reference to this. 
sys.path.append("/opt/daos/lib64/python3.6/site-packages/").

Reply:
@piskvorky What are your thoughts on this? Should we wait for the pydaos thing to be pip-installable before we merge this?

Reply:
I'll be honest, I don't know what DAOS (besides what's in the PR description), not a user. So my only concern is maintenance and I'm -1 on anything non-standard. We're busy enough as it is, not keen on introducing idiosyncrasies.

Reply:
@sridharbalachandriah Speaking of tests, how would you test this reliably? I guess locally you have access to the real system, so you can test against that, but what about in CI? Have a look at how some of the other backends are tested: is it possible to **easily** test DAOS in CI?

Reply:
@mpenkov , I did have a look at the CI's, a fully functional CI would require to emulate the pydaos(DAOS) functionality. I am checking with the DAOS team for ways to address this.

Reply:
@mpenkov , currently DAOS docker container(https://docs.daos.io/v2.2/QSG/docker/) is not published. There is a stand alone mechanism to build DAOS docker container from the rpm, the build has to be  initiated before the DAOS CI tests are invoked. 

Does the smart_open CI infrastructure norms allow for the approach mentioned ?

Reply:
No, not currently.

I think there are two show-stopper problems for this PR:

1) The underlying library is not pip-installable. Yes, it works under the system Python (some _some_ systems), but using the system Python is generally a bad practice, for several reasons. For example, you cannot expect the system's Python version to be what the user wants. So, it's impossible for people to use this contribution in the general case.
2) It's difficult to test this, so the maintenance burden is also high.

@sridharbalachandriah I really appreciate your effort, but I don't think this contribution is a good fit for smart_open right now.

@piskvorky